src/leo_gateway_s3_api.erl

```diff
@@ -508,8 +539,160 @@ handle_1(Req, [{NumOfMinLayers, NumOfMaxLayers}, HasInnerCache, CustomHeaderSett
+    AuthRet_2 = case AuthRet of
+                    {error, Reason} ->
+                        {error, Reason};
+                    {ok, AccessKeyId, _} ->
+                        {ok, AccessKeyId}
+                end,
+    ReqParams_2 = case ReqParams#req_params.is_aws_chunked of
+                      true ->
+                          case AuthRet of
+                              {ok, _, SignParams} ->
+                                  {Signature, SignHead, SignKey} = case SignParams of
+                                                                       undefined ->
+                                                                           {undefined, undefined, undefined};
+                                                                       _ ->
+                                                                           SignParams
+                                                                   end,
+                                  AWSChunkSignParams = #aws_chunk_sign_params{sign_head = SignHead,
+                                                                              sign_key  = SignKey,
+                                                                              prev_sign = Signature,
+                                                                              chunk_sign= <<>>},
+                                  AWSChunkDecState = #aws_chunk_decode_state{buffer         = <<>>,
+                                                                             dec_state      = wait_size,
+                                                                             chunk_offset   = 0,
+                                                                             sign_params    = AWSChunkSignParams,
+                                                                             total_len      = 0},
+                                  ReqParams#req_params{transfer_decode_fun = fun leo_gateway_s3_api:aws_chunk_decode/2, 
+                                                       transfer_decode_state = AWSChunkDecState};
+                              _ ->
+                                  ReqParams
+                          end;
+                      _ ->
+                          ReqParams
+                  end,
+
+    handle_2(AuthRet_2, Req_2, HTTPMethod, Path_1, ReqParams_2, State).
+
+aws_chunk_decode(Bin, State) ->
+    Buffer    = State#aws_chunk_decode_state.buffer,
+    DecState  = State#aws_chunk_decode_state.dec_state,
+    Offset    = State#aws_chunk_decode_state.chunk_offset,
+    SignParams= State#aws_chunk_decode_state.sign_params,
+    TotalLen  = State#aws_chunk_decode_state.total_len,
+    Ret = aws_chunk_decode({ok, <<>>}, <<Buffer/binary, Bin/binary>>, DecState, Offset, SignParams),
+    case Ret of
+        {{error, Reason2}, {_, _, _, _}} ->
+            ?error("aws_chunk_decode", "Parse Error: ~p", [Reason2]),
+            erlang:error("aws-chunked decode failed");
```

Comment:
As it seems you don't handle this error at the cowboy_req:body callers,
LeoFS respond 500 if the signature was wrong. ( should respond 403 according to aws spec )